src/__test__/components/data-management/LaunchAnalysisButton.test.jsx

```diff
@@ -0,0 +1,322 @@
+        'barcodes.tsv.gz': { valid: true, upload: { status: UploadStatus.UPLOADED } },
+        'matrix.mtx.gz': { valid: true, upload: { status: UploadStatus.UPLOADED } },
+      },
+    },
+    [sample2Uuid]: {
+      ...sampleTemplate,
+      name: sample2Name,
+      projectUuid,
+      uuid: sample2Uuid,
+      type: '10X Chromium',
+      metadata: ['value-2'],
+      fileNames: ['features.tsv.gz', 'barcodes.tsv.gz', 'matrix.mtx.gz'],
+      files: {
+        'features.tsv.gz': { valid: true, upload: { status: UploadStatus.UPLOADED } },
+        'barcodes.tsv.gz': { valid: true, upload: { status: UploadStatus.UPLOADED } },
+        'matrix.mtx.gz': { valid: true, upload: { status: UploadStatus.UPLOADED } },
+      },
+    },
+  },
+  backendStatus: {
+    [experiment1id]: {
+      ...initialExperimentBackendStatus,
+      status: {
+        gem2s: {
+          paramsHash: 'old-params-hash',
+          status: PipelineStatus.SUCCEEDED,
+        },
+        pipeline: {
+          status: PipelineStatus.SUCCEEDED,
+        },
+      },
+    },
+  },
+};
+
+describe('LaunchAnalysisButton', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    jest.doMock();
+  });
+
+  it('Process project button is disabled if not all sample metadata are inserted', () => {
+    const notAllMetadataInserted = {
+      ...withDataState,
+      samples: {
+        ...withDataState.samples,
+        [sample1Uuid]: {
+          ...withDataState.samples[sample1Uuid],
+          metadata: [''],
+        },
+      },
+    };
+
+    render(
+      <Provider store={mockStore(notAllMetadataInserted)}>
+        <LaunchAnalysisButton />
+      </Provider>,
+    );
+
+    const button = screen.getByText('Process project').closest('button');
+
+    expect(button).toBeDisabled();
+  });
+
+  it('Launch analysis button is disabled if there is no data', () => {
+    render(
+      <Provider store={mockStore(noDataState)}>
+        <LaunchAnalysisButton />
+      </Provider>,
+    );
+
+    const button = screen.getByText('Process project').closest('button');
+
+    expect(button).toBeDisabled();
+  });
+
+  it('Launch analysis button is disabled if not all data are uploaded', () => {
+    const notAllDataUploaded = {
+      ...withDataState,
+      samples: {
+        ...withDataState.samples,
+        [sample1Uuid]: {
+          ...withDataState.samples[sample1Uuid],
+          files: {
+            ...withDataState.samples[sample1Uuid].files,
+            'features.tsv.gz': { valid: true, upload: { status: UploadStatus.UPLOADING } },
+          },
+        },
+      },
+    };
+
+    render(
+      <Provider store={mockStore(notAllDataUploaded)}>
+        <LaunchAnalysisButton />
+      </Provider>,
+    );
+
+    const button = screen.getByText('Process project').closest('button');
+
+    expect(button).toBeDisabled();
+  });
+
+  it('Launch analysis button is enabled if there is data and all metadata for all samples are uplaoded', () => {
```

Comment:
Rename `Launch analysis` to `Process project`.

Reply:
Renamed